lib/runners/changes.rb

```diff
@@ -47,44 +47,28 @@ def include?(issue)
       end
     end
 
-    def self.calculate(base_dir:, head_dir:, patches:)
-      return calculate_by_patches(head_dir, patches) if patches
-
-      changed_paths = []
-      unchanged_paths = []
-
-      head_dir.glob("**/*", File::FNM_DOTMATCH).filter(&:file?).each do |head_path|
-        relative_path = head_path.relative_path_from(head_dir)
-        base_path = base_dir / relative_path
+    def self.calculate(working_dir:)
+      new(changed_paths: all_files_in(working_dir).sort,
+          unchanged_paths: [],
+          patches: nil)
+    end
 
-        if base_path.file?
-          head_digest = Digest::SHA1.new.file(head_path.to_s)
-          base_digest = Digest::SHA1.new.file(base_path.to_s)
-          if head_digest == base_digest
-            unchanged_paths << relative_path
-          else
-            changed_paths << relative_path
-          end
-        else
-          changed_paths << relative_path
-        end
-      end
+    def self.calculate_by_patches(working_dir:, patches:)
+      all_files = all_files_in(working_dir)
+      changed_files = patches.files.map { |f| Pathname(f) }
 
-      new(changed_paths: changed_paths.sort!,
-          unchanged_paths: unchanged_paths.sort!,
+      new(changed_paths: changed_files.sort,
+          unchanged_paths: (all_files - changed_files).sort,
           patches: patches)
     end
 
-    def self.calculate_by_patches(working_dir, patches)
-      all_files = working_dir
-        .glob("**/*", File::FNM_DOTMATCH)
-        .filter_map { |f| f.relative_path_from(working_dir) if f.file? }
-
-      changed_paths = patches.files.map { |f| Pathname(f) }
-
-      new(changed_paths: changed_paths.sort!,
-          unchanged_paths: (all_files - changed_paths).sort!,
-          patches: patches)
+    def self.all_files_in(basedir)
+      basedir.glob("**/*", File::FNM_DOTMATCH).filter_map do |file|
+        if file.file? && !file.fnmatch?("**/.git/**", File::FNM_DOTMATCH)
```

Comment:
Don't we have to exclude files in the subdirectory of `.git`?

```suggestion
        if file.file? && !file.fnmatch?("**/.git/**/*", File::FNM_DOTMATCH)
```

```irb
irb(main):001:0> Pathname('.').glob('**/.git/**', File::FNM_DOTMATCH).size
=> 15
irb(main):002:0> Pathname('.').glob('**/.git/**/*', File::FNM_DOTMATCH).size
=> 1934
```

Reply:
Good catch! But, `**/.git/**/*` does not match `.git/config`... 😓 
I'll reconsider this logic, thanks. 👍

Reply:
Fixed via e322efd8. 🎉 

Correct pattern: `**/.git/*`

Reply:
:thinking: Does this differ among system environments? My environment matches with `.git/config`.

```ruby
irb(main):001:0> Pathname('.').glob('**/.git/**/*', File::FNM_DOTMATCH).include?(Pathname('.git/config'))
=> true
```

Reply:
Because behavios of `glob` and `fnmatch?` differ:

```
irb(main):005:0> Pathname('.git/config').realpath.fnmatch?('**/.git/*', File::FNM_DOTMATCH)
=> true
irb(main):006:0> Pathname('.git/info/refs').realpath.fnmatch?('**/.git/*', File::FNM_DOTMATCH)
=> true
```

Reply:
Thank you. I misread the method as `glob` (but `fnmatch?` has been used actually).